From f7948e355d85624c577d0fa63977b3bed4d6024f Mon Sep 17 00:00:00 2001
From: Peter Harris <pharris@opentext.com>
Date: Fri, 20 May 2016 19:13:34 -0400
Subject: [PATCH 3/4] res: Fix QueryClientIds reply size

The specification disagrees with itself, so use the part of the
specification that matches the other implementations.

Reviewed-by: Ran Benita <ran234@gmail.com>
Signed-off-by: Peter Harris <pharris@opentext.com>
---
 src/res.xml | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/res.xml b/src/res.xml
index 17e6f83..be32ca1 100644
--- a/src/res.xml
+++ b/src/res.xml
@@ -55,7 +55,15 @@ authorization from the authors.
         <field type="ClientIdSpec" name="spec" />
         <field type="CARD32" name="length" />
         <list type="CARD32" name="value">
-            <fieldref>length</fieldref>
+            <!-- The specification says that the length is in units of CARD32,
+                 but the specification also says that the length is 4 when a
+                 single LocalClientPid is present (ie. the length is in bytes).
+                 The current server implementation sets the length to 4 when a
+                 single CARD32 is present on the wire (length is in bytes). -->
+            <op op="/">
+                <fieldref>length</fieldref>
+                <value>4</value>
+            </op>
         </list>
     </struct>
 
-- 
2.8.3

